### DISCUSSION TITLE
How to override default `motion-safe`

### DISCUSSION THREAD

**Discussion by peteragarclover** [tailwindlabs/tailwindcss#14441] [Help]
Hi everyone! I love that this place exists 😍 

I found a discussion (https://github.com/tailwindlabs/tailwindcss/discussions/11679) that asked and answered how to override default `focus-visible:` modifier by using `addVariant` to redefine it.

I want to do something similar to redefine `motion-safe:` to also check for a class. Our use case is we want to disable smooth scrolling for automated tests, so `motion-safe:smooth-scroll` should only enable smooth scroll if _both_ `prefers-reduced-motion: no-preference` and also `body:not(.prefers-reduced-motion)`. A similar use case could be that you want this to be user configured. I.e. I want this feature to work similar to overriding dark mode selector https://tailwindcss.com/docs/dark-mode#overriding-the-dark-variant.

I extended the playground link from the above discussion about `focus-visible:` and used `addVariant` to try to override `motion-safe`, however... I was not able to override the default. I could only add something with a different name like `motion-safe-new:`

```js
      // This doesn't work
      addVariant(
        'motion-safe',
        '@media (prefers-reduced-motion: no-preference) { body:not(.prefers-reduced-motion) & }',
      )
      // This works - have to use a different name
      addVariant(
        'motion-safe-new',
        '@media (prefers-reduced-motion: no-preference) { body:not(.prefers-reduced-motion) & }',
      )
```

Is there a way to override the default? I worry that other people in our team will accidentally use the default documented one, and not know about custom alternative.

Demo in playground: https://play.tailwindcss.com/fO2j67HXoP?file=config
Credit to @wongjn for creating the playground that this was extended from and answering the original question.

**Comment by JvanderHeide** [comment]
We're looking to use a similar approach albeit for contrast-more (user toggle on the app, as well as using the device preference).
So far I've not been able to either override or disable the builtin variant plugin. Any luck on your side so far?

**Reply by peteragarclover** [reply]
Didn't find a solution. We stuck with adding a different variant with a different name.
